### DISCUSSION TITLE
Adding flags to configure arguments

### DISCUSSION THREAD

**Discussion by bradbell** [coin-or/coinbrew#97] [Q&A] [answered]
If in the coinbrew Ipopt directory you can execute
```
Ipopt>./configure --help | grep java
  --disable-java          disable building of Java interface
Ipopt>
```
How do I pass this flag to the Ipopt configure script ?
```
./coinbrew --help | grep 'pass'
             --xxx=yyy (will be passed through to configure)
             VAR=xxx (will be passed through to configure)
```
I tried adding `--disable-java` to the coinbrew command line but that did not work. I was able to edit the Ipoot/configure script, between the fetch and build, to accomplish what I want as follows:
```
 sed -i -e 's|enable_java="[^"]*"|enable_java=no|' Ipopt/configure
```

**Comment by tkralphs** [comment]
Adding `--disable-java` to the coinbrew command line is how you do it and it seems to get passed through properly. Can you check whether it actually got passed through correctly? Just the fact that it didn't work is not enough to determine if this is indeed the issue. You can just attach `config.log` from the Ipopt subdirectory if you like. For me, the build actually failed out of the box, but this happens with or without the configure argument.

**Comment by bradbell** [comment]
You are correct and below is a script that I used to check that. I do not know what my mistake was.
Thanks.
```
#! /usr/bin/env bash
set -e -u
#
# version
version=3.13.2
#
# n_job
if which nproc >& /dev/null
then
   n_job=$(nproc)
else
   n_job=$(sysctl -n hw.ncpu)
fi
echo "n_job=$n_job"
#
# test_coinbrew
if [ ! -e 'test_coinbrew' ]
then
   mkdir test_coinbrew
fi
cd test_coinbrew
#
# prefix
prefix="$(pwd)/prefix"
#
# coinbrew
if [ ! -e coinbrew ]
then
   web='https://raw.githubusercontent.com/coin-or/coinbrew/master/coinbrew'
   wget $web
   chmod +x coinbrew
fi
#
# Ipopt
if [ ! -e Ipopt ]
then
   ./coinbrew fetch Ipopt@$version --no-prompt
fi
#
# build and install
./coinbrew build Ipopt@$version \
      --prefix="$prefix" \
      --disable-java \
      --parallel-jobs $n_job \
      --verbosity 3 \
      --no-prompt
#
# config.log
grep 'disable-java'  build/Ipopt/$version/config.log
#
echo 'test_coinbrew.sh: OK'
```

**Reply by svigerske** [reply]
Ran this script on my machine, and Ipopt's Java interface is not build.

Ipopt's config.log says, it was called as
```
path/to/test_coinbrew/Ipopt/configure --disable-dependency-tracking --prefix=/path/to/test_coinbrew/prefix --with-coin-instdir=/path/to/test_coinbrew/prefix --disable-java
```

No IpStdJInterface in
```
make[2]: Entering directory '/home/stefan/tmp/test_coinbrew/build/Ipopt/3.13.2/src/Interfaces'
  CXX      IpInterfacesRegOp.lo
  CXX      IpIpoptApplication.lo
  CXX      IpSolveStatistics.lo
  CXX      IpStdCInterface.lo
  CXX      IpStdInterfaceTNLP.lo
  CC       IpStdFInterface.lo
  CXX      IpTNLPAdapter.lo
  CXX      IpTNLPReducer.lo
  CXXLD    libipopt.la
```

And not in tests:
```
Skip testing Java Example (Java interface not build)
```

So we need more details on what "but that did not work" means exactly.

**Comment by bradbell** [comment, marked as answer]
Sorry I was not clearer in the previous message.  Passing --disable-java through coinbrew is working for me. I thought because the documentation had --xxx=yyy that it would not pass through unless there was an equal sign after the flag. I was wrong. Sorry about that.

**Reply by tkralphs** [reply]
Indeed, the documentation is not very clear on that point. I will update it.
